artificer/src/test/java/net/technolords/tools/artificer/input/ArtifactResourceVisitorTest.java

```diff
@@ -14,16 +23,64 @@
+    @DataProvider(name = "dataSetClassifyResource")
+    public Object[][] dataSetClassifyResource() {
+        return new Object[][] {
+                {"ArtificerImplTest.class","D:/commit2Learn/techno-tools/artificer/target/test-classes/net/technolords/tools/artificer/"},
+                {"abc","D:/commit2Learn/techno-tools/artificer/src/test/resources/data/"}
+        };
+    }
+
+    @Test(dataProvider = "dataSetClassifyResource")
+    public void testClassificationOfResource(String input, String location) {
+
+        Path pathToResourceLocation = FileSystems.getDefault().getPath(location);
+        LOGGER.debug("The path towards the input file '" + input + "' exists: " + Files.exists(pathToResourceLocation));
+
+        Analysis analysis = new Analysis();
+        analysis.setArtifactName(input);
+
+        // Create a resource reference linking to the file
+        Resource resource = new Resource();
+        resource.setPath(pathToResourceLocation);
+        resource.setName(input);
+
+        ArtifactResourceVisitor artifactResourceVisitor = new ArtifactResourceVisitor(analysis);
+        artifactResourceVisitor.classifyResource(resource);
+
+        Assert.assertTrue(true);
     }
 
     /**
      * Test case #2: Test adding a resource to a resource group.
      */
-    @Test
-    public void testAddResourceToClassificationGroup() {
-        LOGGER.info("Todo: implementation of test case");
+    @DataProvider(name = "dataSetResourceGroupClassification")
+    public Object[][] dataSetResourceGroupClassification() {
+        return new Object[][] {
+                {"ArtificerImplTest.class","D:/commit2Learn/techno-tools/artificer/target/test-classes/net/technolords/tools/artificer/",".class"},
+                {"abc","D:/commit2Learn/techno-tools/artificer/src/test/resources/data/","_classification_undefined_"},
+                {"artificer-1.0.0-SNAPSHOT.jar","D:/commit2Learn/techno-tools/artificer/src/test/resources/data/jars",".jar"}
+        };
     }
+
+    @Test(dataProvider = "dataSetResourceGroupClassification")
+    public void testAddResourceToClassificationGroup(String input, String location, String classification) {
+
+        Path pathToResourceLocation = FileSystems.getDefault().getPath(location);
+        LOGGER.debug("The path towards the input file '" + input + "' exists: " + Files.exists(pathToResourceLocation));
+
+        Analysis analysis = new Analysis();
+        analysis.setArtifactName("artificer-1.0.0-SNAPSHOT.jar");
+
+        // Create a resource reference linking to the file
+        Resource resource = new Resource();
+        resource.setPath(pathToResourceLocation);
+        resource.setName(input);
+
+        ArtifactResourceVisitor artifactResourceVisitor = new ArtifactResourceVisitor(analysis);
+        artifactResourceVisitor.addResourceToClassificationGroup(resource,classification);
+
+        Assert.assertTrue(analysis.getResourceGroups().containsKey(classification));
+
```

Comment:
Consider to introduce an annotation (see http://testng.org/doc/documentation-main.html#annotations) with the purpose of keeping state. Meaning to say, that if you run a single test with multiple data sets, that TestNG as framework allows you to have fine control of when certain code blocks in your test class gets called. Inside the code blocks you can define properties which can hold state, with the purpose of doing additional assertions. The latter is useful especially when you know the underlying implementation you are testing is keeping state for groups. In other words, if you instantiate an Analysis object once, and use a data set of certain size, the same same can be observed somewhere in the Analysis reference...